feat: Add some screenshots to the App Store Connect setup docs (#4100)
getsentry/sentry-docs: src/platforms/apple/common/dsym.mdx

When you use the App Store Connect integration, Sentry will automatically discover and fetch dSYM files for Bitcode builds as they become available on App Store Connect. These dSYM files however contain obfuscated symbol names and paths. An additional BCSymbolMap file needs to be uploaded to provide properly symbolicated crash reports. These BCSymbolMap files can be uploaded using either [sentry-cli](#symbolmap-sentrycli) or [Fastlane](#symbolmap-fastlane).

> To configure the integration, go to **[Project] > Settings > Debug Files** and add App Store Connect as a new "Custom Repository".

Reviewer: We use quotation marks in the text to quote UI text, so it has to be an exact match. You can also use the phrasing you had before and remove the capital letters and the quotation marks (so: a new custom repository).
```suggestion
To configure the integration, go to **[Project] > Settings > Debug Files** and add App Store Connect in the "Custom Repositories" section.
```

Author: i've also added quotes around "App Store Connect" since it refers to a specific item in the add repositories dropdown.

> ![Getting to a project's settings](getting-to-project-settings.png)

Reviewer: Getting to the project settings is a pretty straightforward direction and adding this particular screenshot doesn't add more information for the reader, so I'd remove this one.  
```suggestion
```

![Adding an App Store Connect Integration to a project](adding-asc-repository.png)

> ![A fully set-up App Store Connect Integration](asc-repository.png)
> 
> Once the integration is set up it will ensure that new builds are detected and fetched within an hour of being available on App Store Connect.

Reviewer: Since the screenshot is of a fully set up integration, it makes sense to put it after the text about a fully set up integration.

```suggestion

Once the integration is configured, it will ensure that new builds are detected and fetched within an hour of being available on App Store Connect.
![A fully configured App Store Connect Integration](asc-repository.png)
```

We highly recommend that you create a dedicated App Store Connect user account for the purpose of this integration, and that you **not** use a regular user account.

Using the configured App Store Connect account on a different device will disrupt Sentry's ability to use it in order to download dSYM files.

</Alert>

These credentials will need to be verified by a two-factor authentication token that you can choose to have sent by SMS.

**Re-authentication**

> The App Store Connect credentials and two-factor authentication are short-lived and need to be refreshed on a regular basis. Sentry will prompt users to re-authenticate in the Project Settings side bar as well as the Custom Repositories section under a project's Debug Files settings once their App Store Connect credentials expire.

Reviewer 2: It is also shown in issue details of the project where the source is configured, example:
![image](https://user-images.githubusercontent.com/146720/132010413-bbbea67b-91f6-4869-91db-7e2564456b0d.png)

> The App Store Connect credentials and two-factor authentication are short-lived and need to be refreshed on a regular basis. Sentry will prompt users to re-authenticate in the Project Settings side bar as well as the Custom Repositories section under a project's Debug Files settings once their App Store Connect credentials expire.

Reviewer: @flub @relaxolotl - does this work?

```suggestion
The App Store Connect credentials and two-factor authentication are short-lived and need to be refreshed on a regular basis. Sentry will prompt users to re-authenticate in the **Issues Details** pages of a project, in the settings menu of a project, as well as in the "Custom Repositories" section of a project's "Debug Files" settings once their App Store Connect credentials expire.
```

Author: looks good to me! i'm wondering if it makes sense to use the singular form for "issue" here because you're looking at one issue instead of several (or a summary of several) issues in floris's example. thoughts?

Reviewer: yep that makes sense. i updated my suggestion

Author: suggestion applied 🎉

> ![Warning to update credentials in sidebar](sidebar-warning.png)

Reviewer 2: This seems way to large.  could it be scaled down?

Reviewer: Don't show the entire sidebar for this. We've already told the reader where to expect to see this warning, so the screenshot is ultimately about showing them what the warning looks like. I'd cut this down to about 5 items in the menu with Debug Files in the middle.

Author: so i can fix this up by simply cropping the existing image, but i can't seem to get the warning to show up any more when the credentials are expired:

<img width="1098" alt="Screenshot 2021-09-07 at 11 09 46 AM" src="https://user-images.githubusercontent.com/5597345/132388039-35e85ed5-9fad-4d18-b664-48bae42df7a3.png">

@flub is this expected behaviour? we can remove everything related to this warning if it's no longer one of the places where we warn the user.

Reviewer 2: this should still be shown so leave it in.  maybe some condition wasn't triggered?  You need to also have pending downloads before it shows up.

> ![Warning to update credentials in Custom Repositories](custom-repositories-warning.png)

Reviewer: Add a space between images
```suggestion

![Warning to update credentials in Custom Repositories](custom-repositories-warning.png)
```
